tools/src/main/java/org/apache/kafka/trogdor/workload/ConsumeBenchWorker.java

```diff
@@ -86,48 +89,91 @@ public void start(Platform platform, WorkerStatusTracker status,
         @Override
         public void run() {
             try {
-                HashSet<TopicPartition> partitions = new HashSet<>();
+                Map<String, List<TopicPartition>> partitionsByTopic = new HashMap<>();
                 for (Map.Entry<String, PartitionsSpec> entry : spec.activeTopics().materialize().entrySet()) {
+                    List<TopicPartition> partitions = new ArrayList<>();
+                    String topic = entry.getKey();
                     for (Integer partitionNumber : entry.getValue().partitionNumbers()) {
-                        partitions.add(new TopicPartition(entry.getKey(), partitionNumber));
+                        partitions.add(new TopicPartition(topic, partitionNumber));
                     }
+                    partitionsByTopic.put(topic, partitions);
                 }
-                log.info("Will consume from {}", partitions);
-                executor.submit(new ConsumeMessages(partitions));
+
+                AbstractConsumeMessages consumeMessagesTask;
+                if (spec.consumerGroup() == null) {
```

Comment:
We don't use null entries in JSON, because it gets too confusing.  You should check against empty string here.

Reply:
oh yeah, I could just set it to an empty string when it's null in the spec - way better

Reply:
Using empty string for `null` is more confusing no?

Reply:
There's a pattern for all of the Trogdor JSON code where we don't use null anywhere.  The problem with null is it gets annoying to check each collection for empty vs. null, each string for empty vs. null, etc. etc.

null is also handled kind of inconsistently in Jackson.  Sometimes Jackson will serialize a field that is null as `"foo": null` whereas sometimes it will just omit the field.  (I think that `"foo": null` is actually not conforming JSON, by the way...)  There are probably ways to configure all this, but null doesn't really provide any value 99% of the time, so it's simpler to just treat empty as null.